homeassistant/components/xiaomi_aqara/config_flow.py

```diff
@@ -55,42 +62,68 @@ async def async_step_user(self, user_input=None):
         if user_input is not None:
```

Comment:
This whole tree is indented and nested making is hard to read.

Turn this into a guard, `if user_input is None:`

Then the whole code below can be flattened more.

Reply:
@frenck I can't because the first time this gets called it needs to skip to the bottom and present the form.
However if an error occurs somewhere in the code inside the `if user_input is not None:` it schould also go back and execute that same code at the bottom to show the form.

Otherwise I would need to move this code in a seprate function and call it both at the bottom and at the top in the guard:
```
        schema = GATEWAY_CONFIG
        if (self.host is None and self.sid is None) or errors:
            schema = GATEWAY_CONFIG_HOST

        return self.async_show_form(step_id="user", data_schema=schema, errors=errors)
```

Maybe I am not understanding you correctly and you ment something else?

Reply:
Right 👍

Reply:
I just made a commit to make it more readable, was that what you ment?
Or schould I revert that commit?